libs/ledger-live-common/src/families/celo/specs.ts

```diff
@@ -1,14 +1,30 @@
-import invariant from "invariant";
-import type { Transaction } from "./types";
-import { getCryptoCurrencyById, parseCurrencyUnit } from "../../currencies";
-import { pickSiblings } from "../../bot/specs";
-import type { AppSpec } from "../../bot/types";
 import { DeviceModelId } from "@ledgerhq/devices";
+import { getCryptoCurrencyById } from "../../currencies";
+import {
+  createLockMutation,
+  createRegisterAccountMutation,
+  createSend50PercentMutation,
+  createSendMaxMutation,
+  createUnlockMutation,
+  createVoteMutation,
+  createActivateVoteMutation,
+  createRevokeVoteMutation,
+  createWithdrawMutation,
+} from "./specs/index";
+import type { AppSpec } from "../../bot/types";
+import type { Transaction } from "./types";
 import { acceptTransaction } from "./speculos-deviceActions";
 
 const currency = getCryptoCurrencyById("celo");
-const minimalAmount = parseCurrencyUnit(currency.units[0], "0.001");
-const maxAccount = 3;
+const send50PercentMutation = createSend50PercentMutation();
+const sendMaxMutation = createSendMaxMutation();
+const registerAccountMutation = createRegisterAccountMutation();
+const unlockMutation = createUnlockMutation();
+const lockMutation = createLockMutation();
+const voteMutation = createVoteMutation();
+const activateVoteMutation = createActivateVoteMutation();
+const revokeVoteMutation = createRevokeVoteMutation();
+const withdrawMutation = createWithdrawMutation();
```

Comment:
what is the idea behind these functions? as celo would be the first to adopt this pattern, i'm not sure this helps readability and future maintenance. also what happens to the "maxAccount" shared value then?

alternatively you could define all the mutation as local vailable in that same file, and then compose your mutations:[...] as you did 🤔

Reply:
Hey @gre - I can explain some of the decisions here.  

Regarding the factory functions that return mutations, I put these in separate files for readability and maintainability.  Breaking apart the mutations into separate files was, to me, a superior to having a 1000 line spec file.  

Regarding sharing variables, that contradicts the direction I was given - @haammar-ledger indicated that each mutation _should_ be able to live on it's own and not be dependent on other mutations (see discussion in Discord [here](https://discord.com/channels/885256081289379850/950694044802097172/1002249261800493107)).

Reply:
Readability is a matter of taste :slightly_smiling_face: I also don't mind to have this code split like that.
LGTM, closing this thread, feel free to reopen.